### ISSUE TITLE
docs:  Need Answer to Actual Explanation of when to use Bloc and Cubit 

### ISSUE THREAD

**Issue by ShivamMattoo33** [felangel/bloc#3169]
Hi @felangel  I had some questions which I am struggling to get concrete answers to

1. A cubit can do everything a bloc can do except handle streams. Streams are used in our app in only 4-5 places. How do we know when to use Cubit and not Bloc?
2. Is it possible to define a standard for using Blocs or Cubits( eg. Use Blocs when end goal is Api calls but for everything else use Cubits)?
3. If Cubits offer better performance  and less boilerplate why not use them for everything except for streams?
4. With Bloc updates reducing boilerplate code is there any need for Cubits in the future?

**Comment by alvinvin00** [comment]
1. Cubit and Bloc are now 90% identical, basically Cubit is a Bloc without event transformer and traceability, I suggest you to start from Cubit and refactor later to Bloc if you have to
2. You can mix and match it since they extends BlocBase, i have done it in my app
3. I don't think there's a performance penalty when using Bloc, and you're allowed to use all Cubits in your app if you want to.
4. Blocs and Cubits while fundamentally similar, served different purposes. Cubits are for simplicity, Blocs are for traceability (in my opinion)

**Comment by felangel** [comment]
Hi @ShivamMattoo33 👋 
Thanks for opening an issue!

Have you checked out the [Bloc vs. Cubit](https://bloclibrary.dev/#/coreconcepts?id=cubit-vs-bloc) section in the docs?

**Comment by felangel** [comment]
> Hello @felangel Thank you so much for replying

No problem! 😄 

Yes I have read through this section and I understand Simplicity vs Traceability I guess let me rephrase my questions into 3 simple Questions basically to know your opinion
> 
> 1. If In our Team we wanna make a decision between Whether to use **Cubit** or **Bloc** what would you recommend(**Just want to know your Opinion**)?. To give you more context, 90% of our code can work with both Cubits and Bloc as it doesn't use Steams.

If you don't envision leveraging events for logging, analytics, etc. then I'd recommend using cubit since it's the simpler solution.

> 2. Do you think it is a good idea to have **mix of both** in a codebase?. The problem I see fore-coming is that for example if both can do the same Job It will come down to opinions within team as there is no **Standard** for using either one as they both would work for the same thing.

I think it's fine to mix them as long as there are clear reasons for when to use one vs the other. For example, in the [Fluttersaurus](https://github.com/felangel/fluttersaurus) example we use a bloc for the search functionality (because we need to be able to transform the incoming event stream via restartable/debounce) and we use a cubit for the synonyms functionality because it doesn't require any of the event sourcing or event transformation functionality.

> 3. With the Bloc updates and lesser boilerplate Do you think Cubits are not needed/ redundant and could be consumed into Bloc?

It really just depends. For example, at VGV we tend to use blocs across the board for consistency and because in most cases we do take advantage of the events for things like analytics and developer logs. I personally don't think the decision to use one vs the other should be made based on the number of lines of code (as you mentioned it's very similar) but rather based on whether you benefit from the added event stream that bloc provides.

Let me know if that helps and thanks for the great questions!

**Comment by ShivamMattoo33** [comment]
That answers it Thanks @felangel
